docs/en/app-server/build.md

```diff
@@ -124,3 +124,100 @@ Where:
 
 ### Known limitation
 - At the moment only GitHub and GitLab repositories are supported.
+
+
+# Embedding a Server
+
+In some cases, it can be useful to embed a RoadRunner server inside another GO program. This is often the case in microservice architectures where you may have a mandated GO framework for all the apps. In such cases it might not be possible to run a stock roadrunner instance and the only choice is to run roadrunner inside the main app framework / program.
+
+This is now possible in version 2.11 and newer. Let's pretend we have a GO app with an HTTP handler and we want to pass the request to PHP via RoadRunner running in the same go app.
+
+```go
+func handleRequest(w http.ResponseWriter, request *http.Request) {
+    // Find a way to pass that to RoadRunner so PHP handles the request
+}
+```
+
+## Create an RR instance
+
+```go
+overrides := []string{} // List of configuration overrides
+plugins := roadrunner.DefaultPluginsList() // List of RR plugins to enable
+rr, err := roadrunner.NewRR(".rr.yaml", &overrides, plugins)
+```
+
+Here we use the default list of plugins. The same list of plugin you would get if you were to run `rr serve` with a stock roadrunner binary. 
+
+You can however chose only the plugins you want and add your own private plugins as well:
+
+```go
+overrides := []string{} // List of configuration overrides
+plugins := []interface{}{
+    &informer.Plugin{},
+    &resetter.Plugin{},
+    // ...
+    &httpPlugin.Plugin{},
+    // ...
+    &coolCompany.Plugin{},
+}
+rr, err := roadrunner.NewRR(".rr.yaml", &overrides, plugins)
+```
+
+## Passing requests to RoadRunner
+
+Roadrunner can respond to HTTP requests, but also gRPC ones or many more. Because this is all done via plugins that each listen to different types of requests, ports, etc...
+
+So when we talk about passing a request to roadrunner, we're actually talking about passing the request to roadrunner's HTTP plugin. To do this, we need to keep a handle on the http plugin.
+
+```go
+overrides := []string{} // List of configuration overrides
+httpPlugin := &httpPlugin.Plugin{},
+plugins := []interface{}{
+    &informer.Plugin{},
+    &resetter.Plugin{},
+    // ...
+    httpPlugin,
+    // ...
+    &coolCompany.Plugin{},
+}
+rr, err := roadrunner.NewRR(".rr.yaml", &overrides, plugins)
+```
+
+The HTTP plugin is itself an `http.Handler` so it's now very easy to use it to let roadrunner and PHP handle the request:
+
+
+```go
+overrides := []string{} // List of configuration overrides
+httpPlugin := &httpPlugin.Plugin{},
+plugins := []interface{}{
+    &informer.Plugin{},
+    &resetter.Plugin{},
+    // ...
+    httpPlugin,
+    // ...
+    &coolCompany.Plugin{},
+}
+rr, err := roadrunner.NewRR(".rr.yaml", &overrides, plugins)
```

Comment:
After RR PR merge, `&overrides` can be passed as a value. There is no need to pass slices as a reference since a slice contains a reference to data under the hood. So, you won't copy a whole slice, but only one internal pointer. (+ len and cap ints).